libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/roadlabel/RoadLabelSurfaceLayer.kt

```diff
@@ -53,34 +57,38 @@ class RoadLabelSurfaceLayer(
     override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
         logAndroidAuto("RoadLabelSurfaceLayer carMapSurface loaded")
         super.onAttached(mapboxCarMapSurface)
-        mapboxCarMapSurface.mapSurface.getMapboxMap().getStyle { style ->
-            val aboveLayer = style.styleLayers.last().id.takeUnless {
-                it == BELOW_LAYER
+        styleLoadedListener = mapboxCarMapSurface.handleStyleOnAttached { style ->
+            if (!persistentStyleAdded) {
```

Comment:
We don't need to have this `persistentStyleAdded`

The persisted layer can be added once. But then we need to update the bitmap every time the style changes

```
        mapboxCarMapSurface.getStyle { style ->
            val aboveLayer = style.styleLayers.last().id.takeUnless {
                it == BELOW_LAYER
            }
            style.addPersistentStyleCustomLayer(
                layerId = CAR_NAVIGATION_VIEW_LAYER_ID,
                carTextLayerHost,
                LayerPosition(aboveLayer, BELOW_LAYER, null)
            ).error?.let {
                logAndroidAutoFailure("Add custom layer exception $it")
            }
        }
        styleLoadedListener = mapboxCarMapSurface.handleStyleOnAttached {
            val bitmap = roadLabelRenderer.render(
                roadNameObserver.currentRoad,
                roadNameObserver.currentShields,
                roadLabelOptions()
            )
            carTextLayerHost.offerBitmap(bitmap)
        }
```

Reply:
^ This is looking to fix this issue https://github.com/mapbox/mapbox-navigation-android/pull/5853#issuecomment-1145048674

The `carTextLayerHost.offerBitmap(bitmap)` needs to be called when the style changes